rustls/src/client/client_conn.rs

```diff
@@ -232,17 +234,41 @@ impl Clone for ClientConfig {
 }
 
 impl ClientConfig {
+    /// Create a builder for a client configuration with the default
+    /// [`CryptoProvider`]: [`crate::crypto::ring::default_provider`] and safe ciphersuite and
+    /// protocol defaults.
+    ///
+    /// For more information, see the [`ConfigBuilder`] documentation.
     #[cfg(feature = "ring")]
+    pub fn builder() -> ConfigBuilder<Self, WantsVerifier> {
+        // Safety: we know the *ring* provider's ciphersuites are compatible with the safe default protocol versions.
+        Self::builder_with_provider(crate::crypto::ring::default_provider().into())
+            .with_safe_default_protocol_versions()
+            .unwrap()
+    }
+
     /// Create a builder for a client configuration with the default
-    /// [`CryptoProvider`]: [`crate::crypto::ring::default_provider`].
+    /// [`CryptoProvider`]: [`crate::crypto::ring::default_provider`], safe ciphersuite defaults and
+    /// the provided protocol versions.
     ///
     /// For more information, see the [`ConfigBuilder`] documentation.
-    pub fn builder() -> ConfigBuilder<Self, WantsVersions> {
+    #[cfg(feature = "ring")]
+    pub fn builder_with_protocol_versions(
+        versions: &[&'static versions::SupportedProtocolVersion],
+    ) -> ConfigBuilder<Self, WantsVerifier> {
+        // Safety: we know the *ring* provider's ciphersuites are compatible with all protocol version choices.
```

Comment:
moreover: we know that the provider has at least one ciphersuite for each version

i think this will panic for `builder_with_protocol_versions(&[])`; i don't think that necessarily needs fixing.

Reply:
> i think this will panic for builder_with_protocol_versions(&[]); i don't think that necessarily needs fixing.

I added a note to each fn's docstring about this :+1: